src/lib.rs

```diff
@@ -4541,6 +4545,82 @@ pub trait Itertools: Iterator {
 
 impl<T> Itertools for T where T: Iterator + ?Sized {}
 
+/// An [`ExactSizeIterator`] blanket implementation that provides extra adaptors
+/// and methods.
+///
+/// This trait defines a number of adapators take an iterator and parameter as
+/// input, and return a new iterator value. An example of an adaptor is
+/// [`.zip_squash()`](ExactSizeItertools::zip_squash)
+pub trait ExactSizeItertools: ExactSizeIterator {
+    /// Create an iterator which iterates over both this and the specified
+    /// iterator simultaneously, yielding pairs of elements.
+    ///
+    /// Yields elements evenly sampled from the longest iterator.
+    ///
+    /// ```ignore
+    /// use itertools::ExactSizeItertools;
+    /// let it = vec![0,1].iter().zip_squash(vec![1,2,3].iter());
+    /// itertools::assert_equal(it, vec![(0,1),(1,3)]);
+    /// ```
+    fn zip_squash<J>(self, other: J) -> ZipSquash<Self, J::IntoIter>
+    where
+        J: IntoIterator,
+        <J as IntoIterator>::IntoIter: ExactSizeIterator,
+        Self: Sized;
```

Comment:
Couldn't this method just be on `Itertools` with a `Self: ExactSizeIterator` bound? Why introduce a new trait?

Reply:
Oh yeah I can do that. Updated to do that.